server/routes/river-and-sea-levels.js

```diff
@@ -1,28 +1,63 @@
 const joi = require('@hapi/joi')
 const ViewModel = require('../models/views/river-and-sea-levels')
+const RiverViewModel = require('../models/views/river-stations')
 const floodService = require('../services/flood')
 const locationService = require('../services/location')
+const util = require('../util')
 
 module.exports = [{
   method: 'GET',
   path: '/river-and-sea-levels',
   handler: async (request, h) => {
-    const { q: location } = request.query
-    var model, place, stations
-    place = await locationService.find(location)
-    if (typeof place === 'undefined' || place === '') {
+    const { type } = request.query
+    let { q: location } = request.query
+
+    let model, place, stations
+
+    if (type === 'location') {
+      place = await locationService.find(util.cleanseLocation(location))
+
+      if (typeof place === 'undefined' || place === '') {
+        model = new ViewModel({ location, place, stations, type })
+        model.referer = request.headers.referer
+        return h.view('river-and-sea-levels', { model })
+      }
+      if (!place.isEngland.is_england) {
+        return h.view('location-not-england')
+      }
+
+      stations = await floodService.getStationsWithin(place.bbox)
+
+      model = new ViewModel({ location, place, stations, type })
+      model.referer = request.headers.referer
+
+      return h.view('river-and-sea-levels', { model })
+    } else if (type === 'river') {
+      location = util.cleanseLocation(location)
+
+      const rivers = floodService.rivers
+      const isRiver = rivers.some(obj => obj === location)
+
+      if (isRiver === false || location === '') {
+        model = new ViewModel({ location, place, stations, type })
+        model.referer = request.headers.referer
+        return h.view('river-and-sea-levels', { model })
+      }
+      stations = await floodService.getStationsByRiver(util.cleanseLocation(location))
+
+      model = new RiverViewModel({ location, place, stations, type })
+      model.referer = request.headers.referer
+
+      return h.view('river-and-sea-levels', { model })
+    } else {
+      // getStationsWithin needs changing
```

Comment:
This service needs changing to a new one that gets all stations, I think this was something lazy i put in when getting the code running. Can you amend to `// TODO: ...` so that I can pick it up later on and change to get the data from the cache